src/commands/signzone.rs

```diff
@@ -1418,41 +1419,67 @@ impl SignZone {
         Ok(records)
     }
 
-    fn find_apex(
-        records: &SortedRecords<StoredName, StoredRecordData, MultiThreadedSorter>,
-    ) -> Result<(StoredName, Class, Ttl, Serial), Error> {
-        let soa = match records.find_soa() {
-            Some(soa) => soa,
-            None => {
-                return Err(Error::from("Invalid zone file: Cannot find SOA record"));
+    // TODO: Add tests.
+    fn find_apex<'a>(
```

Comment:
@Philip-NLnetLabs: Hang on. We discussed yesterday that we probably don't even want this functionality as discovering the apex is prone to problems, the apex should be the origin found during zone parsing shouldn't it?

Reply:
Zone parsing doesn't find the origin if the zone has no $ORIGIN and only absolute names. I don't think $ORIGIN has to be the name of the zone. You could have $ORIGIN com. and then example SOA xxxx. So the zone name would be example.com.

Reply:
Good point. It's confusing that both `ldns-signzone` and `dnssec-signzone` use `-o <origin>` to refer to what is actually I believe the zone name, not the `$ORIGIN` in a zone file.

Reply:
The confusing part is that -o is optional. It should be required, then the problem is solved.